### DISCUSSION TITLE
required property

### DISCUSSION THREAD

**Discussion by laszlo-zaptic** [lit/lit#2628] [Q&A] [answered]
how to make a property required? something like
```typescript
    @property({required: true})
    placeholder: string
```
so that TS and/or runtime error appear if this property isn't set?

**Comment by justinfagnani** [comment, marked as answer]
This is actually an existing lit-analyzer feature request: https://github.com/runem/lit-analyzer/issues/74

cc @rictic

**Reply by laszlo-zaptic** [reply]
i'm using the component with react also, and in @lit-labs/react this line seems to make every property and event optional
https://github.com/lit/lit/blob/b8ceafb00651b9a4b5021ac5a3d1d960ceb10e6b/packages/labs/react/src/create-component.ts#L134

would it make sense to define properties for the class as optional like
```typescript
 @property()
 placeholder?: string
```
and then the properties (maybe also events) could be removed from the `Partial<>`?
